### ISSUE TITLE
im2col! incompatibility with Complex numbers

### ISSUE THREAD

**Issue by hiiroo** [denizyuret/Knet.jl#500]
im2col! unable to process arrays with Complex{} types. Note that, I have faced this issue only on the cpu side. 

Some sample code to test
```
using AutoGrad
using Knet
test_input = rand(Complex{Float32}, 6, 6, 1, 1)
cw = xavier(Complex{Float32}, 3, 3, 1, 5)
ow = xavier(Complex{Float32}, 2, 80)
function f(x)
    x = conv4(cw, x)
    x = ow*mat(x)
end
f(test_input)
```

and expected error

```ERROR: MethodError: no method matching im2col!(::Array{Complex{Float32},4}, ::Array{Complex{Float32},4}, ::Array{Complex{Float32},2}, ::Int64, ::Int64, ::Int64, ::Int64, ::Int64, ::Int64)```

**Comment by denizyuret** [comment]
Works in the latest master, which should be released as v1.2.7 in the next couple days (or you can get it now with `using Pkg; pkg"dev Knet"`. Of course this does not mean all other functions will work, keep testing...

**Comment by hiiroo** [comment]
I'm closing the issue, since it has solved with the new release.
